src/devices/Ina219/Ina219.cs

```diff
@@ -0,0 +1,372 @@
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Device;
+using System.Device.I2c;
+
+namespace Iot.Device.Adc
+{
+    /// <summary>
+    /// Binding that exposes an INA219 Bidirectional Current/Power monitor.
+    /// 
+    /// The INA219 is a high-side current shunt and power monitor with an I2C interface.
+    /// The INA219 monitors both shunt drop and supply voltage, with programmable conversion
+    /// times and filtering. A programmable calibration value, combined with an internal multiplier,
+    /// enables direct readouts in amperes. An additional multiplying register calculates power in watts.
+    /// <see href="http://www.ti.com/lit/ds/symlink/ina219.pdf"/>
+    /// </summary>
+    public class Ina219 : IDisposable
+    {
+        private I2cDevice _i2cDevice;
+        private bool _disposeI2cDevice = false;
+        private ushort _calibrationValue;
+        private Ina219AdcResolutionOrSamples _busAdcResSamp;
+        private Ina219AdcResolutionOrSamples _shuntAdcResSamp;
+        private float _currentLsb;
+
+        // These are possitive ranges in volts associated with each gain setting of the Programable Gain Controller.
+        private readonly Dictionary<Ina219PgaSensitivity, float> _sensitivityVoltage = new Dictionary<Ina219PgaSensitivity, float>()
+        { 
+            { Ina219PgaSensitivity.PlusOrMinus40mv, 0.04F },
+            { Ina219PgaSensitivity.PlusOrMinus80mv, 0.08F },
+            { Ina219PgaSensitivity.PlusOrMinus160mv, 0.16F },
+            { Ina219PgaSensitivity.PlusOrMinus320mv, 0.32F }
+        };
+
+        // These values are the datasheet defined delays in micro seconds between requesting a Current or Power value from the INA219 and the ADC sampling having completed
+        // along with any conversions.
+        private readonly Dictionary<Ina219AdcResolutionOrSamples, int> _readDelays = new Dictionary<Ina219AdcResolutionOrSamples, int>()
+        {
+            { Ina219AdcResolutionOrSamples.Adc9Bit, 84 },
+            { Ina219AdcResolutionOrSamples.Adc10Bit, 148 },
+            { Ina219AdcResolutionOrSamples.Adc11Bit, 276 },
+            { Ina219AdcResolutionOrSamples.Adc12Bit, 532 },
+            { Ina219AdcResolutionOrSamples.Adc2Sample, 1006 },
+            { Ina219AdcResolutionOrSamples.Adc4Sample, 2130 },
+            { Ina219AdcResolutionOrSamples.Adc8Sample, 4260 },
+            { Ina219AdcResolutionOrSamples.Adc16Sample, 8510 },
+            { Ina219AdcResolutionOrSamples.Adc32Sample, 17020 },
+            { Ina219AdcResolutionOrSamples.Adc64Sample, 34050 },
+            { Ina219AdcResolutionOrSamples.Adc128Sample, 68100 }
+        };
+
+        /// <summary>
+        /// Method to initialize values during device construction
+        /// </summary>
+        /// <param name="i2cDevice"></param>        
+        private void _Ina219(I2cDevice i2cDevice)
```

Comment:
call it `Initialize`, we usually don't use underscores in the method names

Reply:
@krwq 

All suggested changes done and PR updated.